doc/port_auto_neg/port-auto-negotiation-design.md

```diff
@@ -84,8 +92,13 @@ Currently, SAI already defines a few port attributes to support port auto negoti
 1. SAI implementation must return error code if any of the auto negotiation related attribute is not supported, swss and syncd must not crash.
 2. SAI implementation must keep backward compatible. As long as swss and SAI keep backward compatible, user need not change anything after this feature is enabled in SONiC.
 3. If autoneg is enabled and adv_speeds is not configured or empty, SAI must advertise it with all supported speeds.
-4. If autoneg is enabled and adv_interface_types is not configured or empty, SAI must advertise it with all supported interface types.
-5. If autoneg is disabled and interface_type is not configured, SAI must use SAI_PORT_INTERFACE_TYPE_NONE.
+4. If autoneg is enabled and adv_speeds is specified, SAI must advertise only the speeds those are supported by the switch silicon. Hence the operational advertisement could be a subset of the speeds specified in adv_speeds. SAI should return errors if none of desirable advertised speeds is valid.
+5. If autoneg is enabled and adv_interface_types is not configured or empty, SAI must advertise it with all supported interface types.
+6. If autoneg is enabled and adv_interface_types is specified, SAI must advertise only the interface types those are valid to the attched transceiver and supported by the switch silicon. Hence the operational advertisement could be a subset of the interface types specified in adv_interface_types. SAI should return errors if none of the desirable advertised interface types is valid.
+7. If autoneg is enabled, the administrative port speed updates via SAI_PORT_ATTR_SPEED should disable the autoneg.
+8. If autoneg is enabled, the interface type updates via SAI_PORT_ATTR_INTERFACE_TYPE should be ignored. These are requests from xcvrd triggered by the transceiver insertion if the per-port interface type is specified in the platform-specific media_settings.json.
+9. If autoneg is enabled on a SFP port, SAI should also activate the link-training to dynamically tune the TX FIR (i.e. Clause 73).
+10. If autoneg is disabled, the port speed, interface type and TX FIR should be restored to the original values before autoneg activation.
```

Comment:
What are the original values before autoneg activation? What if user has not configured these.

Reply:
if autoneg is disabled then if speed/fec/if_type were configured they should be set, if not configured – default values should be set explicitly. This is to ensure the hardware is not programmed with residual values of the autoneg

Reply:
Thanks for the comments, this will be updated as follow
`11. If autoneg is disabled, the port speed, FEC, interface type and TX FIR should be restored with "speed", "fec", "interface_type" and ("preemphasis", "idriver", "ipredriver", "pre1", "pre2", "pre3", "main", "post1", "post2", "post3", "attn") values in the APPL_DB. If the corresponding configuration is not specified in the APPL_DB, vendor-specific SAI driver defaults should be applied.`

Reply:
"vendor-specific SAI driver defaults should be applied" - This should not be the case. It should be SAI API defaults and not any vendor specific defaults to have a uniformity of the behavior.

Reply:
Done, the HLD is now updated, please check it out and see if works